app/login/page.tsx

```diff
@@ -1,49 +1,93 @@
-import Link from 'next/link';
-import { signIn } from '@/lib/auth';
+"use client";
 
-import { SimpleLayout } from '@/components/page-layouts';
-import { Label } from '@/components/ui/label';
-import { Button } from '@/components/ui/button';
-import { Input } from '@/components/ui/input';
+import Link from "next/link";
+import { signInFromLogin } from "@/lib/actions";
+
+import { SimpleLayout } from "@/components/page-layouts";
+import { Label } from "@/components/ui/label";
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+import { useForm, SubmitHandler } from "react-hook-form";
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
 
 export default function Login() {
+  const schema = z.object({
+    email: z.string().min(1, { message: "Email is required" }).email(),
+    password: z.string().min(1, { message: "Password is required" }),
+  });
+
+  type FormFields = z.infer<typeof schema>;
+
+  const {
+    register,
+    handleSubmit,
+    setError,
+    formState: { errors },
+  } = useForm<FormFields>({
+    resolver: zodResolver(schema),
+  });
 
-  const signInAction = async (formData: FormData) => {
-    'use server';
-    await signIn('credentials', {
-      redirectTo: '/',
-      email: formData.get('email') as string,
-      password: formData.get('password') as string,
+  const onSubmit: SubmitHandler<FormFields> = async (data) => {
+    const ret = await signInFromLogin({
+      redirectTo: "/",
+      email: data.email,
+      password: data.password,
     });
+    console.log(ret);
```

Comment:
same as the register onSubmit, we could add some UI to tell the user that the login failed or succeeded and is being redirected